public/js/paystack-forms-public.js

```diff
@@ -1,676 +1,677 @@
+        });
+      });
+      var international_card = false;
       if ($("#pf-vamount").length) {
         var amountField = $("#pf-vamount");
+        calculateTotal();
       } else {
-        var amountField = $("#pf-qamount");
+        var amountField = $("#pf-amount");
       }
-      var unit = amountField.val();
-      var quant = $("#pf-quantity").val();
-
-      if (quant == "" || quant == null) {
-        quant = 1;
+      var max = 10;
+      amountField.keydown(function(e) {
+        format_validate(max, e);
+      });
+  
+      amountField.keyup(function(e) {
+        checkMinimumVal();
+      });
+  
+      function checkMinimumVal() {
+        if ($("#pf-minimum-hidden").length) {
+          var min_amount = Number($("#pf-minimum-hidden").val());
+          var amt = Number($("#pf-amount").val());
+          if (amt < min_amount) {
+            $("#pf-min-val-warn").text(
+              "Amount cannot be less than the minimum amount"
+            );
+          } else {
+            $("#pf-min-val-warn").text("");
+          }
+        }
       }
-      var newvalue = unit * quant;
-      $("#pf-amount").val(newvalue);
-    }
-    function calculateFees(transaction_amount) {
-      setTimeout(function() {
-        transaction_amount = transaction_amount || parseInt(amountField.val());
-        if ($("#pf-vamount").length) {
-          var name = $("#pf-vamount option:selected").attr("data-name");
-          $("#pf-vname").val(name);
+  
+      function format_validate(max, e) {
+        var value = amountField.text();
+        if (e.which != 8 && value.length > max) {
+          e.preventDefault();
         }
+        // Allow: backspace, delete, tab, escape, enter and .
         if (
-          transaction_amount == "" ||
-          transaction_amount == 0 ||
-          transaction_amount.length == 0 ||
-          transaction_amount == null ||
-          isNaN(transaction_amount)
+          $.inArray(e.keyCode, [46, 8, 9, 27, 13, 110, 190]) !== -1 ||
+          // Allow: Ctrl+A
+          (e.keyCode == 65 && e.ctrlKey === true) ||
+          // Allow: Ctrl+C
+          (e.keyCode == 67 && e.ctrlKey === true) ||
+          // Allow: Ctrl+X
+          (e.keyCode == 88 && e.ctrlKey === true) ||
+          // Allow: home, end, left, right
+          (e.keyCode >= 35 && e.keyCode <= 39)
         ) {
-          var total = 0;
-          var fees = 0;
-        } else {
-          var obj = new KkdPffPaystackFee();
-
-          obj.withAdditionalCharge(settings.fee.adc);
-          obj.withThreshold(settings.fee.ths);
-          obj.withCap(settings.fee.cap);
-          obj.withPercentage(settings.fee.prc);
-
-          var total = obj.addFor(transaction_amount * 100) / 100;
-          var fees = total - transaction_amount;
+          // let it happen, don't do anything
+          calculateFees();
+          return;
         }
-        $(".pf-txncharge")
-          .hide()
-          .html("NGN" + fees.toFixed(2))
```

Comment:
@ibrahimlawal how do we turn off this feature for other currencies since they have different pricing?